### ISSUE TITLE
Compile on save vs build

### ISSUE THREAD

**Issue by abogartz** [TypeStrong/atom-typescript#222]
Perhaps my config is just not set up correctly, but I have my tsconfig file set up as follows.
```
"version": "1.4.1",
    "compilerOptions": {
        "target": "es5",
        "declaration": false,
        "out": "main.js",
        "noImplicitAny": false,
        "removeComments": false,
        "noLib": false,
        "sourceMap": false
    },

    "files": [
        "./main.ts"

    ]
```
Main.ts has references in it to other classes, e.g "a.ts" with class A.

If I make a change to a.ts, the "save on compile" command triggers, but the resulting js file doesn't seem to have all of the Typescript code such as the __extends function. However, if I run a "build" from the package, then the resulting js is correct.

**Comment by basarat** [comment]
We don't have good support for out as I don't personally use or recommend using it.

Tracked here : https://github.com/TypeStrong/atom-typescript/issues/206

**Comment by abogartz** [comment]
That's too bad. Just out of curiosity, if you don't use --out, then what's your workflow for using multiple .ts files? If you're using Atom, how do you keep a list of js files current in your HTML markup?

**Comment by csnover** [comment]
@abogartz 

> if you don't use --out, then what's your workflow for using multiple .ts files?

Use external modules.

> how do you keep a list of js files current in your HTML markup?

Don’t. Use an AMD loader.

**Comment by basarat** [comment]
@abogartz I've documented my reasons for not doing `--out` : https://github.com/TypeStrong/atom-typescript/blob/master/docs/out.md 

Based on that I'll marking this explicitly as `wont-fix`, in #206 and also in `tsconfig.md`.

We can provide an `--out` story using some external build tools later.
